### PR TITLE
Impl multi currency trait

### PR DESCRIPTION
**Pull request by jordy25519**
Reinstate this trait. It's very useful for CENNZX and anything that wants multiple currencies

### CODE REVIEW

File: prml/generic-asset/src/imbalances.rs [hunk 1]
```diff
@@ -36,14 +36,14 @@ use sp_std::{mem, result};
...
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
 pub struct PositiveImbalance<T: Trait> {
 	amount: T::Balance,
 	asset_id: T::AssetId,
 }
 
 impl<T: Trait> PositiveImbalance<T> {
-	/// Create a new positive imbalance given some amount
+	/// Create a new negative imbalance from a `balance` and with the given `asset_id`.
```
Comment (jordy25519) [inline prml/generic-asset/src/imbalances.rs]: will fix this

File: prml/generic-asset/src/impls.rs [hunk 2]
```diff
@@ -0,0 +1,696 @@
+// Copyright 2019 Plug New Zealand Ltd.
```
Comment (KarishmaBothara) [inline prml/generic-asset/src/impls.rs]: 2019-2020

File: prml/generic-asset/src/impls.rs [hunk 3]
```diff
@@ -0,0 +1,696 @@
...
+use sp_runtime::{
+	traits::{AtLeast32BitUnsigned, CheckedSub, MaybeSerializeDeserialize, Zero},
+	DispatchError, DispatchResult,
+};
+use sp_std::{fmt::Debug, result};
+
+// Note: in the following traits the terms:
+// - 'token' / 'asset' / 'currency' and
+// - 'balance' / 'value' / 'amount'
+// are used interchangeably as they make more sense in certain contexts.
```
Comment (KarishmaBothara) [inline prml/generic-asset/src/impls.rs:31]: 👍 nice explanation

File: prml/generic-asset/src/impls.rs [hunk 4]
```diff
@@ -0,0 +1,696 @@
...
+
+				// Make free balance be equal to `amount` again
+				let _ = <GenericAsset as MultiCurrencyAccounting>::make_free_balance_be(alice, None, amount);
+				assert_eq!(
+					<GenericAsset as MultiCurrencyAccounting>::free_balance(alice, None),
+					amount
+				);
+				assert_eq!(GenericAsset::total_issuance(16001), amount);
+
+				// Mint `amount` of `asset_id` into `alice`s account. Similar to `deposit_creating` above
```
Comment (KarishmaBothara) [inline prml/generic-asset/src/impls.rs]: // Mint `amount` of default currency into `alice`s account. Similar to `deposit_creating` above

**Comment by jordy25519** [comment]
ci error is just network timeout during build

**Review by KarishmaBothara — APPROVED** [review]

**Pull request merged**